docs/examples/setup-and-solve.rst

```diff
@@ -101,59 +101,49 @@ C
     #include "osqp.h"
 
     int main(int argc, char **argv) {
-        // Load problem data
+        /* Load problem data */
         c_float P_x[3] = {4.0, 1.0, 2.0, };
-        c_int P_nnz = 3;
-        c_int P_i[3] = {0, 0, 1, };
-        c_int P_p[3] = {0, 1, 3, };
-        c_float q[2] = {1.0, 1.0, };
+        c_int   P_nnz  = 3;
+        c_int   P_i[3] = {0, 0, 1, };
+        c_int   P_p[3] = {0, 1, 3, };
+        c_float q[2]   = {1.0, 1.0, };
         c_float A_x[4] = {1.0, 1.0, 1.0, 1.0, };
-        c_int A_nnz = 4;
-        c_int A_i[4] = {0, 1, 0, 2, };
-        c_int A_p[3] = {0, 2, 4, };
-        c_float l[3] = {1.0, 0.0, 0.0, };
-        c_float u[3] = {1.0, 0.7, 0.7, };
+        c_int   A_nnz  = 4;
+        c_int   A_i[4] = {0, 1, 0, 2, };
+        c_int   A_p[3] = {0, 2, 4, };
+        c_float l[3]   = {1.0, 0.0, 0.0, };
+        c_float u[3]   = {1.0, 0.7, 0.7, };
         c_int n = 2;
         c_int m = 3;
 
-        // Exitflag
-        c_int exitflag = 0;
+        /* Exitflag */
+        c_int exitflag;
 
-        // Workspace structures
+        /* Workspace, settings, matrices */
         OSQPWorkspace *work;
-        OSQPSettings  *settings = (OSQPSettings *)c_malloc(sizeof(OSQPSettings));
-        OSQPData      *data     = (OSQPData *)c_malloc(sizeof(OSQPData));
-
-        // Populate data
-        if (data) {
-            data->n = n;
-            data->m = m;
-            data->P = csc_matrix(data->n, data->n, P_nnz, P_x, P_i, P_p);
-            data->q = q;
-            data->A = csc_matrix(data->m, data->n, A_nnz, A_x, A_i, A_p);
-            data->l = l;
-            data->u = u;
-        }
-
-        // Define solver settings as default
-        if (settings) {
-            osqp_set_default_settings(settings);
-            settings->alpha = 1.0; // Change alpha parameter
-        }
-
-        // Setup workspace
-        exitflag = osqp_setup(&work, data, settings);
-
-        // Solve Problem
+        OSQPSettings *settings;
+        csc *P, *A;
+
+        /* Populate matrices */
+        P = csc_matrix(n, n, P_nnz, P_x, P_i, P_p);
+        A = csc_matrix(m, n, A_nnz, A_x, A_i, A_p);
+
+        /* Settings.  Change alpha parameter */
+        settings = (OSQPSettings *)c_malloc(sizeof(OSQPSettings));
+        if (settings) osqp_set_default_settings(settings);
+        settings->alpha = 1.0;
```

Comment:
I have not modified any of the code, examples or documentation in ./docs, since I think it is better to agree first whether this refactor is a good approach.

Reply:
Noted as a v1.0 update [here](https://github.com/oxfordcontrol/osqp/projects/1#column-1967224).

Reply:
Noted as a v1.0 update [here](https://github.com/oxfordcontrol/osqp/projects/1#column-1967224).